stdlib/Dates/src/types.jl

```diff
@@ -203,7 +203,7 @@ function totaldays(y, m, d)
 end
 
 # If the year is divisible by 4, except for every 100 years, except for every 400 years
-isleapyear(y) = (y % 4 == 0) && ((y % 100 != 0) || (y % 400 == 0))
+isleapyear(y::Integer) = (y % 4 == 0) && ((y % 100 != 0) || (y % 400 == 0))
```

Comment:
An alternative would be to change it to
```suggestion
isleapyear(y:) = iszero(y % 4) && (!iszero(y % 100) || iszero(y % 400))
```
which would work for `Year(1992)`. Alas, then someone might come with another date-like object that again triggers a similar issue... so perhaps best to restrict it to `Integer`

(explicit type annotations like this are good for avoiding many type invalidation issues, so I think this is a good idea anyway)